File: codeforces/44.candies_and_two_sisters_[1335A]/cats.py
# Author: José Rodolfo (jric2002)
import math
t = int(input())
for i in range(0, t):
  n = int(input())
  r = 0
  if ((n % 2) == 0):
    r = int(n / 2) - 1
  else:
    r = math.ceil(n / 2) - 1
  print(r)

File: codeforces/low_difficulty/difficulty_800/44.candies_and_two_sisters_[1335A]/cats.cpp
#include <iostream>
#include <cmath>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned int t;
  unsigned long int n, r;
  cin >> t;
  for(unsigned int i = 0; i < t; i++) {
    cin >> n;
    if ((n % 2) == 0) {
      r = (n / 2) - 1;
    }
    else {
      r = ceil((float)n / 2.0) - 1;
    }
    cout << r << endl;
  }
  return 0;
}